packages/contracts-bedrock/test/L2/CrossL2Inbox.t.sol

```diff
@@ -0,0 +1,154 @@
+    CrossL2Inbox crossL2Inbox;
+
+    /// @dev Sample ICrossL2Inbox.Identifier.
+    ICrossL2Inbox.Identifier sampleIdentifier = ICrossL2Inbox.Identifier({
+        origin: address(0),
+        blocknumber: 0,
+        logIndex: 0,
+        timestamp: block.timestamp,
+        chainId: block.chainid
+    });
+
+    /// @dev Sets up the test suite.
+    function setUp() public {
+        crossL2Inbox = new CrossL2Inbox();
+    }
+
+    /// @dev Tests that `executeMessage` succeeds when called with valid parameters.
+    function testFuzz_executeMessage_succeeds(
+        bytes calldata _msg,
+        ICrossL2Inbox.Identifier calldata _id,
+        address _target,
+        uint256 _value
+    )
+        external
+        payable
+    {
+        vm.assume(_id.timestamp <= block.timestamp);
+        vm.assume(_target.code.length == 0);
+
+        // need to prevent call to L1Block.isInDependencySet from reverting
+        vm.mockCall({
+            callee: Predeploys.L1_BLOCK_ATTRIBUTES,
+            data: abi.encodeWithSelector(L1Block.isInDependencySet.selector, _id.chainId),
+            returnData: abi.encode(true)
+        });
+
+        vm.deal(tx.origin, _value);
+
+        // executeMessage
+        vm.prank(tx.origin);
+        vm.expectCall(_target, _value, _msg);
+        crossL2Inbox.executeMessage{ value: _value }({ _id: _id, _target: _target, _msg: _msg });
+
+        assertEq(crossL2Inbox.origin(), _id.origin);
+        assertEq(crossL2Inbox.blocknumber(), _id.blocknumber);
+        assertEq(crossL2Inbox.logIndex(), _id.logIndex);
+        assertEq(crossL2Inbox.timestamp(), _id.timestamp);
+        assertEq(crossL2Inbox.chainId(), _id.chainId);
+    }
+
+    /// @dev Tests that `executeMessage` fails when called with an identifier with an invalid timestamp.
+    function test_executeMessage_invalidTimestamp_fails() external {
+        sampleIdentifier.timestamp = block.timestamp + 1;
+
+        vm.prank(tx.origin);
+        vm.expectRevert("CrossL2Inbox: invalid id timestamp");
+        crossL2Inbox.executeMessage({ _id: sampleIdentifier, _target: address(0), _msg: hex"1234" });
+    }
+
+    /// @dev Tests that `executeMessage` fails when called with an identifier with an invalid chain ID.
+    function test_executeMessage_invalidChainId_fails() external {
+        vm.mockCall({
+            callee: Predeploys.L1_BLOCK_ATTRIBUTES,
+            data: abi.encodeWithSelector(L1Block.isInDependencySet.selector, sampleIdentifier.chainId),
+            returnData: abi.encode(false)
+        });
+
+        vm.prank(tx.origin);
+        vm.expectRevert("CrossL2Inbox: id chain not in dependency set");
+        crossL2Inbox.executeMessage({ _id: sampleIdentifier, _target: address(0), _msg: hex"1234" });
+    }
+
+    /// @dev Tests that `executeMessage` succeeds when called with an identifier with the same chain ID as
+    ///      the current chain.
+    function test_executeMessage_sameChainId_succeeds() external {
+        vm.mockCall({
```

Comment:
This test isn't super useful because it isn't executing the code that would run in production due to the mocked call, altho i do think its good to think of this test case

Reply:
mainly the idea is to check whether `CrossL2Inbox` picks up that it's the same chain ID, abstracted from whatever that number may be (so if it works for our testing suite, it should work for mainnet)